src/impls/primitive.rs

```diff
@@ -116,14 +230,41 @@ macro_rules! ImplDekuRead {
 
                 let (bit_slice, rest) = input.split_at(bit_size);
 
-                let bytes: &[u8] = bit_slice.as_raw_slice();
+                let pad = 8 * ((bit_slice.len() + 7) / 8) - bit_slice.len();
+
+                // TODO: mention that this is slow? a conditional should be fast...
```

Comment:
fyi, I don't think this path is slow, I think it's more-so the allocation that we save with the specialization of bytes (try running test_alloc with the changes in this change on master, we save an alloc + realloc)

Reply:
I agree